ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionAnalyzer.java

```diff
@@ -57,10 +67,75 @@ public void analyzeInternal(ASTNode ast) throws SemanticException {
     inputs.add(new ReadEntity(table));
 
     ShowPartitionsDesc desc = new ShowPartitionsDesc(tableName, ctx.getResFile(), partSpec);
+    analyzeShowPartitionsConstraints(ast, table, desc);
     Task<DDLWork> task = TaskFactory.get(new DDLWork(getInputs(), getOutputs(), desc));
     rootTasks.add(task);
 
     task.setFetchSource(true);
     setFetchTask(createFetchTask(ShowPartitionsDesc.SCHEMA));
   }
+
+  private void analyzeShowPartitionsConstraints(ASTNode ast, Table tab,
```

Comment:
Please don't use long methods like this at ddl. Instead  cut it to 3 different methods, similar to getPartitionSpecs, each retrieving one data, returning it to analyzeInternal. Also please don't use setters at ShowPartitionDesc, a desc object at ddl should have final variables set at the constructor, and not by setters. This declares that they may not change later. Thanks.

Reply:
@miklosgergely Thanks for reviewing. Update codes to move the parser into 3 different methods and make the variable in ShowPartitionDesc all final.